src/app/core/services/asset-property-mapper.ts

```diff
@@ -27,76 +28,133 @@ export class AssetPropertyMapper {
-    );
-    const transportMode = lookup(AssetProperties.transportMode, (id) =>
-      this.transportModeSelectItemService.findById(id),
-    );
-    const keywords = (props[AssetProperties.keywords] ?? '')
-      .split(',')
-      .map((it) => it.trim())
-      .filter((it) => it);
+    const language = asset.language
+      ? this.languageSelectItemService.findById(asset.language)
+      : null;
+    const dataCategory = asset.dataCategory
+      ? this.dataCategorySelectItemService.findById(asset.dataCategory)
+      : null;
+    const dataSubcategory = asset.dataSubcategory
+      ? this.dataSubcategorySelectItemService.findById(asset.dataSubcategory)
+      : null;
+    const transportMode = asset.transportMode
+      ? this.transportModeSelectItemService.findById(asset.transportMode)
+      : null;
 
-    const id = props[AssetProperties.id] ?? 'no-id-was-set';
-    const additionalProperties = this.buildAdditionalProperties(props);
+    const keywords = asset.keywords ?? [];
+
+    const id = asset.assetId ?? 'no-id-was-set';
+    const additionalProperties = this.buildAdditionalProperties(
+      asset.additionalProperties,
+    );
 
     return {
       id,
-      name: props[AssetProperties.name] ?? id,
-      version: props[AssetProperties.version],
-      contentType: props[AssetProperties.contentType],
+      name: asset.name ?? id,
+      version: asset.version ?? null,
+      contentType: asset.mediaType ?? null,
       originator: opts.connectorEndpoint,
       originatorOrganization:
-        props[AssetProperties.curatorOrganizationName] ??
-        'Unknown Organization',
+        asset.creatorOrganizationName ?? 'Unknown Organization',
       keywords,
-      description: props[AssetProperties.description],
+      description: trimmedOrNull(asset.description),
       language,
-      publisher: props[AssetProperties.publisher],
-      standardLicense: props[AssetProperties.standardLicense],
-      endpointDocumentation: props[AssetProperties.endpointDocumentation],
+      publisher: asset.publisherHomepage ?? null,
+      standardLicense: asset.licenseUrl ?? null,
+      endpointDocumentation: asset.landingPageUrl ?? null,
       dataCategory,
       dataSubcategory,
-      dataModel: props[AssetProperties.dataModel],
-      geoReferenceMethod: props[AssetProperties.geoReferenceMethod],
+      dataModel: asset.dataModel ?? null,
+      geoReferenceMethod: asset.geoReferenceMethod ?? null,
       transportMode,
-      httpProxyMethod: this._parseBoolean(
+      httpProxyMethod: this._parseBoolean(asset.httpDatasourceHintsProxyMethod),
+      httpProxyPath: this._parseBoolean(asset.httpDatasourceHintsProxyPath),
+      httpProxyQueryParams: this._parseBoolean(
+        asset.httpDatasourceHintsProxyBody,
+      ),
+      httpProxyBody: this._parseBoolean(asset.httpDatasourceHintsProxyBody),
+      additionalProperties,
+    };
+  }
+
+  buildUiAssetfromprops(props: Record<string, string>): UiAsset {
```

Comment:
this method shouldnot be required anymore.

the class "AssetProperties" should be removed. in fact this class should be removed, and "Asset" should become:

```ts

type Asset = UiAsset & {connectorEndpoint: string;}

```